src/corefx/System.Globalization.Native/idna.cpp

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include <stdint.h>
+#include <uchar.h>
+#include <unicode/uidna.h>
+
+const uint32_t AllowUnassigned = 0x1;
+const uint32_t UseStd3AsciiRules = 0x2;
+
+uint32_t GetOptions(uint32_t flags)
+{
+    uint32_t options = UIDNA_DEFAULT;
+
+    if ((flags & AllowUnassigned) == AllowUnassigned)
+    {
+        options |= UIDNA_ALLOW_UNASSIGNED;
+    }
+
+    if ((flags & UseStd3AsciiRules) == UseStd3AsciiRules)
+    {
+        options |= UIDNA_USE_STD3_RULES;
+    }
+
+    return options;
+}
+
+/*
+Function:
+ToASCII
+
+Used by System.Globalization.IdnMapping.GetAsciiCore to convert an Unicode domain name to ASCII
+
+Return values:
+0: internal error during conversion.
+>0: the length of the converted string (not including the null terminator).
+*/
+extern "C" int32_t ToAscii(uint32_t flags, const UChar* lpSrc, int32_t cwSrcLength, UChar* lpDst, int32_t cwDstLength)
```

Comment:
I don't see ToAscii and ToUnicode being called from managed code.  I do see IsNormalized and NormalizeString being called.  Will these methods be used in the future?